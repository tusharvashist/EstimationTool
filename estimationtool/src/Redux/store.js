import { configureStore } from '@reduxjs/toolkit'
import loginReducer from './loginRedux'
import basicDetailReducer from './basicDetailRedux'
import effortAttributeSave  from './effortAttributeSaveRedux'
import calcAttributeSave  from './CalcAttributeRedux'

export default configureStore({
  reducer: {
      login: loginReducer,  
      basicDetail: basicDetailReducer, 
      effortAttribute: effortAttributeSave,
      calcAttribute: calcAttributeSave
  },
})