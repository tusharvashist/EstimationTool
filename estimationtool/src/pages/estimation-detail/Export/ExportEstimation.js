import {
  Checkbox,
  FormControl,
  FormControlLabel,
  FormGroup,
  FormLabel,
} from "@mui/material";
import React, { useState } from "react";
import CustomizedDialogs from "../../../shared/ui-view/dailog/dailog";
import EstimationService from "../estimation.service";
import Snackbar from "../../../shared/layout/snackbar/Snackbar";
import { useSelector } from "react-redux";

export const ExportEstimationPopup = (props) => {
  const estimationHeaderId = useSelector((state) => state.estimationHeaderId);

  const [estimationDetail, setEstimationDetail] = useState(true);
  const [estimationSummary, setEstimationSummary] = useState(false);
  const [resourceCount, setResourceCount] = useState(false);
  const [resourcePlanning, setResourcePlanning] = useState(false);
  const [resourceTimeline, setResourceTimeline] = useState(false);

  const [isOpen, setOpen] = useState({});

  const popSubmitHandler = () => {
    callAPI();
  };

  const callAPI = () => {
    const payload = {
      estimationHeaderId: estimationHeaderId.estHeaderId,
      reports: [
        {
          key: "estimationDetail",
          value: estimationDetail,
        },
        {
          key: "estimationSummary",
          value: estimationSummary,
        },
        {
          key: "resourceCount",
          value: resourceCount,
        },
        {
          key: "resourcePlanning",
          value: resourcePlanning,
        },
        {
          key: "resourceTimeline",
          value: resourceTimeline,
        },
      ],
    };

    EstimationService.getAllExportData(payload)
      .then((res) => {
        props.closeExportEstimation();

        if (res.status === 200) {
          EstimationService.getReport();
        }
      })
      .catch((err) => {
        console.log(err);
        // setOpen({
        //   open: true,
        //   severity: "error",
        //   message: err.response.data.message,
        // });
      });
  };

  const handleClose = () => {
    setOpen({});
  };

  const { message, severity, open } = isOpen || {};

  return (
    <>
      <CustomizedDialogs
        isOpen={props.openExport}
        openFun={props.openExportEstimation}
        closeFun={props.closeExportEstimation}
        title={props.title}
        oktitle={props.oktitle}
        cancelTitle={props.cancelTitle}
        saveFun={popSubmitHandler}
        width={"sm"}
        buttonType="submit"
      >
        <form>
          <FormControl component="fieldset" className="form-export">
            <FormLabel component="legend">
              Please select from below options to export:
            </FormLabel>
            <FormGroup>
              <FormControlLabel
                control={<Checkbox defaultChecked />}
                label="Estimation Details"
                onChange={(e) => setEstimationDetail(e.target.checked)}
              />
              <FormControlLabel
                control={<Checkbox />}
                label="Estimation Summary"
                onChange={(e) => setEstimationSummary(e.target.checked)}
              />
              <FormControlLabel
                control={<Checkbox />}
                label="Resource Count"
                onChange={(e) => setResourceCount(e.target.checked)}
              />
              <FormControlLabel
                control={<Checkbox />}
                label="Resource Planning"
                onChange={(e) => setResourcePlanning(e.target.checked)}
              />
              <FormControlLabel
                control={<Checkbox />}
                label="Resource Timeline"
                onChange={(e) => setResourceTimeline(e.target.checked)}
              />
            </FormGroup>
          </FormControl>
        </form>
      </CustomizedDialogs>
      {open && (
        <Snackbar
          isOpen={open}
          severity={severity}
          autoHideDuration={6000}
          onClose={handleClose}
          message={message}
        />
      )}
    </>
  );
};
