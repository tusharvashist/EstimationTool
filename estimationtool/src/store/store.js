import { configureStore } from "@reduxjs/toolkit";
import toasterReducer from "./toasterSlice";

export default configureStore({
  reducer: {
    toaster: toasterReducer,
  },
});
