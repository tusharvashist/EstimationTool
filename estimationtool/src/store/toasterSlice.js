import { createSlice } from "@reduxjs/toolkit";

export const toasterSlice = createSlice({
  name: "toaster",
  initialState: {
    open: false,
    message: "custom message",
    severity: "success",
    duration: 2000,
    positionVertical: "top",
    positionHorizontal: "right",
  },
  reducers: {
    openSuccessToasterWithMsg: (state, action) => {
      state.open = true;
      state.severity = "success";
      if (action.payload.message) {
        state.message = action.payload.message;
      }
      if (action.payload.duration) {
        state.duration = action.payload.duration;
      }
      if (action.payload.positionVertical) {
        state.positionVertical = action.payload.positionVertical;
      }
      if (action.payload.positionHorizontal) {
        state.positionHorizontal = action.payload.positionHorizontal;
      }
    },
    openErrorToasterWithMsg: (state, action) => {
      state.open = true;
      state.severity = "error";
      if (action.payload.message) {
        state.message = action.payload.message;
      }
      if (action.payload.duration) {
        state.duration = action.payload.duration;
      }
      if (action.payload.positionVertical) {
        state.positionVertical = action.payload.positionVertical;
      }
      if (action.payload.positionHorizontal) {
        state.positionHorizontal = action.payload.positionHorizontal;
      }
    },
    openWarningToasterWithMsg: (state, action) => {
      state.open = true;
      state.severity = "warning";
      if (action.payload.message) {
        state.message = action.payload.message;
      }
      if (action.payload.duration) {
        state.duration = action.payload.duration;
      }
      if (action.payload.positionVertical) {
        state.positionVertical = action.payload.positionVertical;
      }
      if (action.payload.positionHorizontal) {
        state.positionHorizontal = action.payload.positionHorizontal;
      }
    },
    openInfoToasterWithMsg: (state, action) => {
      state.open = true;
      state.severity = "info";
      if (action.payload.message) {
        state.message = action.payload.message;
      }
      if (action.payload.duration) {
        state.duration = action.payload.duration;
      }
      if (action.payload.positionVertical) {
        state.positionVertical = action.payload.positionVertical;
      }
      if (action.payload.positionHorizontal) {
        state.positionHorizontal = action.payload.positionHorizontal;
      }
    },
    closeToaster: (state) => {
      state.open = false;
    },
  },
});

export const {
  openSuccessToasterWithMsg,
  openErrorToasterWithMsg,
  openWarningToasterWithMsg,
  openInfoToasterWithMsg,
  closeToaster,
} = toasterSlice.actions;

export default toasterSlice.reducer;
