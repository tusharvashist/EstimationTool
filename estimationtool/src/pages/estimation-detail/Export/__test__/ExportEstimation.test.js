import { ExportEstimationPopup } from "../ExportEstimation";
import { render, screen } from "@testing-library/react";
import "jest-canvas-mock";
import { toBeEnabled, toBeDisabled } from "@testing-library/jest-dom";

import store from "../../../../Redux/store";
import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";
import { BrowserRouter, Route } from "react-router-dom";
import { persistStore } from "redux-persist";

let persistor = persistStore(store);

import Enzyme, { shallow, render as enzymeRender, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

beforeEach(() => {
  let openExport = true;
  const openExportEstimation = () => {
    openExport = true;
  };

  const closeExportEstimation = () => {
    openExport = false;
  };
  const exportFun = () => {};

  render(
    <BrowserRouter>
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <ExportEstimationPopup
            openExport={openExport}
            openExportEstimation={openExportEstimation}
            closeExportEstimation={closeExportEstimation}
            title="Export Estimation"
            oktitle="Generate"
            cancelTitle="Cancel"
            exportFun={exportFun}
          />
        </PersistGate>
      </Provider>
    </BrowserRouter>
  );
});

describe("All check boxes should be enabled and working", () => {
  it("check Estimation Detail checkbox is enabled", () => {
    expect(
      screen.getByRole("checkbox", { name: /Estimation Details/i })
    ).toBeEnabled();
    expect(
      screen.getByRole("checkbox", { name: /Estimation Details/i })
    ).toBeChecked();
  });

  it("check Estimation Summary checkbox is enabled", () => {
    expect(
      screen.getByRole("checkbox", { name: /Estimation Summary/i })
    ).toBeEnabled();
    expect(
      screen.getByRole("checkbox", { name: /Estimation Summary/i })
    ).not.toBeChecked();
  });

  it("check Resource Count checkbox is enabled", () => {
    expect(
      screen.getByRole("checkbox", { name: /Resource Count/i })
    ).toBeEnabled();
    expect(
      screen.getByRole("checkbox", { name: /Resource Count/i })
    ).not.toBeChecked();
  });

  it("check Resource Planning checkbox is enabled", () => {
    expect(
      screen.getByRole("checkbox", { name: /Resource Planning/i })
    ).toBeEnabled();
    expect(
      screen.getByRole("checkbox", { name: /Resource Planning/i })
    ).not.toBeChecked();
  });

  it("check Resource Timeline checkbox is enabled", () => {
    expect(
      screen.getByRole("checkbox", { name: /Resource Timeline/i })
    ).toBeEnabled();
    expect(
      screen.getByRole("checkbox", { name: /Resource Timeline/i })
    ).not.toBeChecked();
  });
});

describe("snapshot testing", () => {
  it("genrate snapshot of Export estimation", () => {
    let openExport = true;
    const openExportEstimation = () => {
      openExport = true;
    };

    const closeExportEstimation = () => {
      openExport = false;
    };
    const exportFun = () => {};

    const wrapper = shallow(
      <BrowserRouter>
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <ExportEstimationPopup />
          </PersistGate>
        </Provider>
      </BrowserRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("Test methods after mount", () => {
  const wrapper = mount(
    <BrowserRouter>
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <ExportEstimationPopup />
        </PersistGate>
      </Provider>
    </BrowserRouter>
  );

  wrapper.instance().expect(wrapper.instance().callAPI()).toHaveBeenCalled();
});
