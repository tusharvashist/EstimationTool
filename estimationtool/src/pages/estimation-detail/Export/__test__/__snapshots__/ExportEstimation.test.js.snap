// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing genrate snapshot of Export estimation 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <PersistGate
      loading={null}
      persistor={
        Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "flush": [Function],
          "getState": [Function],
          "pause": [Function],
          "persist": [Function],
          "purge": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <ExportEstimationPopup />
    </PersistGate>
  </Provider>
</Router>
`;
